### DISCUSSION TITLE
How to close the current buffer using cokeline?

### DISCUSSION THREAD

**Discussion by niamleeson** [noib3/cokeline.nvim#197] [Q&A]
Cokeline is nice since it focuses an available buffer when it closes. I'd like to keymap to close the current buffer. How can I get the current buffer so I can call `buffer:close()`?

**Comment by niamleeson** [comment]
I was able to achieve it like this. Is this recommended?

```
vim.keymap.set('n', '<leader>bd', function()
  local lazy = require 'cokeline.lazy'
  local state = lazy 'cokeline.state'
  local curbufnumber = tonumber(vim.fn.bufnr '%')

  local target_buffer = vim.tbl_filter(function(buffer)
    return buffer.number == curbufnumber
  end, state.visible_buffers)[1]

  if not target_buffer then
    print 'Current buffer not found'
    return
  end

  target_buffer:delete()
end, { desc = 'close buffer' })
```

**Reply by willothy** [reply]
You can use `require('cokeline.buffers').get_current()` to simplify things.

**Reply by niamleeson** [reply]
Thank you!

**Reply by willothy** [reply]
Also there's no need to use `cokeline.lazy` in mappings, that just exists to improve the plugin's startup time. In this case it will just be wasting memory because it creates a table and metatable with each call to `lazy`.

**Reply by dsully** [reply]
Thanks, I needed this too. Perhaps add it to the documentation?
